terraform/modules/files/setup_rancher.sh.tpl

```diff
@@ -3,11 +3,20 @@
 # Wait for Rancher UI to boot
 printf 'Waiting for Rancher to start'
 until $(curl --output /dev/null --silent --head --insecure --fail ${rancher_host}); do
-    printf '.'
-    sleep 5
+	printf '.'
+	sleep 5
 done
 
-sudo apt-get install jq -y || sudo yum install jq -y
+# Wait for apt-get to be unlocked
+printf 'Waiting for apt-get to unlock'
+sudo fuser /var/lib/dpkg/lock >/dev/null 2>&1;
+while [ $? -ne 1 ]; do
+	printf '.';
+	sleep 5;
+	sudo fuser /var/lib/dpkg/lock >/dev/null 2>&1;
+done
+
+sudo apt-get install jq -y
```

Comment:
It looks like this drops support for RedHat-derived distros and anyone else that uses yum instead of apt.  The failure mode on distros that aren't derived from Debian is that it will loop forever, ignoring the `Specified filename /var/lib/dpkg/lock does not exist` error message.

Is there a bounded set of callers that all ensure they are on a system that uses apt and dpkg before this script is called?

Reply:
We never really had support for RedHat based distros. The `|| sudo yum install jq -y` was just a remnant of a previous attempt to support RedHat based distros.

At this point we only support Ubuntu 16.x.

We don't have any guards in place to verify users only use Ubuntu 16.x.

All that being said I don't believe the failure mode would be to loop forever. If the file doesn't exist then fuser will exit with code 1, which will skip the loop entirely.